src/threading/Manager.h

```diff
@@ -143,6 +144,9 @@ class Manager {
     using msg_thread_list = std::list<MsgThread*>;
     msg_thread_list msg_threads;
 
+    using io_source_list = std::list<iosource::IOSource*>;
+    io_source_list io_sources;
```

Comment:
Better to avoid new raw pointers / manual memory mgmt. Seems the manager is owning these, so suggest to use `std::list<std::unique_ptr<iosource::IOSource>>`. That will then take care of the other clean-up pieces as well, like in `terminate()`. 

(I realize we do manual memory mgt for the threads, too, but I believe that's just a left-over from the past and we should move those over to smart pointers as well.